src/webots/gui/WbGuiApplication.cpp

```diff
@@ -381,6 +382,16 @@ bool WbGuiApplication::setup() {
   bool showGuidedTour =
     prefs->value("Internal/firstLaunch", true).toBool() && mStartWorldName.isEmpty() && WbMessageBox::enabled();
 
+  if (!mStream) {  // create streaming server for robot window if not in stream mode.
+    bool monitorActivity = false;
+    bool disableTextStreams = false;
+    bool ssl = false;
+    bool controllerEdit = false;
+    int port = 1234;
+    mStreamingServer = new WbStreamingServer(monitorActivity, disableTextStreams, ssl, controllerEdit, mStream);
+    mStreamingServer->start(port);
```

Comment:
Note: the streaming port should be made available in the Webots preferences.
```suggestion
    mStreamingServer = new WbStreamingServer(false, false, false, false, mStream);
    mStreamingServer->start(1234);
```